numpy/core/tests/test_datetime.py

```diff
@@ -518,31 +518,38 @@ def test_pydatetime_creation(self):
     def test_datetime_string_conversion(self):
         a = ['2011-03-16', '1920-01-01', '2013-05-19']
         str_a = np.array(a, dtype='S')
+        uni_a = np.array(a, dtype='U')
         dt_a = np.array(a, dtype='M')
-        str_b = np.empty_like(str_a)
-        dt_b = np.empty_like(dt_a)
 
         # String to datetime
         assert_equal(dt_a, str_a.astype('M'))
         assert_equal(dt_a.dtype, str_a.astype('M').dtype)
+        dt_b = np.empty_like(dt_a)
         dt_b[...] = str_a
         assert_equal(dt_a, dt_b)
+
         # Datetime to string
         assert_equal(str_a, dt_a.astype('S0'))
+        str_b = np.empty_like(str_a)
         str_b[...] = dt_a
         assert_equal(str_a, str_b)
 
-        # Convert the 'S' to 'U'
-        str_a = str_a.astype('U')
-        str_b = str_b.astype('U')
```

Comment:
This wasn't a good test, as it would still pass if the `str_b[...] = ...` below was a no op!